### PR TITLE
[INLONG-3355][Manager] Check the source states before return group state in manager client

### LINKED ISSUE #3355: [Improve] Check sourceState When return GroupState in Manager Client
### Description

_No response_

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kipshi**
### Title Name: [INLONG-3355] Check the source states before return group state in manager client

Fixes #3355

### Verifying this change

- [X] Make sure that the change passes the CI checks.

### CODE REVIEW

**Review by gong — APPROVED** [review]

**Review by crancchen — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/stream/InlongStreamResponse.java [hunk 1]
```diff
@@ -85,29 +84,6 @@
     @ApiModelProperty(value = "Names of responsible persons, separated by commas")
     private String inCharges;
 
-    @ApiModelProperty(value = "Status")
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/stream/InlongStreamResponse.java:88]: The response needs to show those fields for the user.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/InlongGroupContext.java [hunk 2]
```diff
@@ -91,24 +93,53 @@ private void recheckState() {
...
                 this.groupErrLogs.computeIfAbsent("failedSources", Lists::newArrayList)
                         .add(GsonUtil.toJson(failedSource));
             }
+            return;
+        }
+        // check if any stream source is in indirect state
+        switch (this.state) {
+            case STARTED:
+                for (StreamSource source : sourcesInGroup) {
+                    if (!(source.getState() == State.NORMAL)) {
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/InlongGroupContext.java:125]: Why not use `source.getState() != State.NORMAL`?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractStreamSourceOperation.java [hunk 3]
```diff
@@ -126,11 +126,6 @@ public void updateOpt(SourceRequest request, Integer groupStatus, String operato
         // Setting updated parameters of stream source entity.
         setTargetEntity(request, entity);
         entity.setVersion(entity.getVersion() + 1);
-        if (GroupState.forCode(groupStatus).equals(GroupState.CONFIG_SUCCESSFUL)) {
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractStreamSourceOperation.java:129]: Why remove those lines?

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/InlongGroupContext.java (hunk 2 above)
Comment (kipshi) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/InlongGroupContext.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractStreamSourceOperation.java (hunk 3 above)
Comment (healchow) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractStreamSourceOperation.java:129]: If the source was normal or failed, after the update operation, the status should be transfer to `new` or `to_be_add`.

Comment (kipshi) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractStreamSourceOperation.java:129]: if the source was frozen, then you change to new or to_be_add, the status is fuzed

Comment (kipshi) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractStreamSourceOperation.java:129]: change like this
 // If failed , return to new so can be init again
        if (SourceState.forCode(entity.getStatus()) == SourceState.SOURCE_FAILED){
            entity.setStatus(SourceState.SOURCE_NEW.getCode());
        }

Comment (healchow) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractStreamSourceOperation.java:129]: Should we need concerned about the group status?
For example, if the Group was SUCCESS or FAILED, how to control the Source status?

Comment (healchow) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractStreamSourceOperation.java:129]: We can handle it like this: do not modify the state when modifying the source info, and then call the restart interface after the modification.

Comment (kipshi) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractStreamSourceOperation.java:129]: done

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**